### PR TITLE
Fix Incorrect Tracking Protection State

### PR DESCRIPTION
**Pull request by evancrow**
First, we weren't even updating the tracking state when the URL changed (main issue). The state setting was flakey as we store if tracking is allowed but set the foreground state as protection enabled (confusing).

### How was this tested?
- [x] I tested this manually

## Issues addressed
Fixed: #3521
Fixed: #3534

### CODE REVIEW

File: content-blocker-lib-ios/src/TrackingPreventionConfig.swift [hunk 1]
```diff
@@ -24,11 +24,22 @@ struct TrackingPreventionConfig {
         Defaults[.unblockedDomains].remove(domain)
     }
 
-    static func trackersAllowedFor(_ domain: String) -> Bool {
-        Defaults[.unblockedDomains].contains(domain)
+    static func trackersPreventedFor(_ domain: String) -> Bool {
```
Comment (chung-lai) [inline content-blocker-lib-ios/src/TrackingPreventionConfig.swift:27]: I think we have this concept of whitelisting a domain and allow trackers for them. Is renaming this to `PreventedFor` make it easier to understand in the code? I found `trackersAllowedFor` easier to read somehow 😅

**Review by chung-lai — APPROVED** [review]
LGTM. Just left one feedback on renaming. Thanks for fixing this. Wonder if we are confident about this change or merge this after branch for some bake time of the change.

File: content-blocker-lib-ios/src/TrackingPreventionConfig.swift (hunk 1 above)
Comment (evancrow) [reply content-blocker-lib-ios/src/TrackingPreventionConfig.swift:27]: Makes sense, will change it back!

Comment (evancrow) [reply content-blocker-lib-ios/src/TrackingPreventionConfig.swift:27]: Actually I'll keep both, make them two separate functions...

**Comment by evancrow** [comment]
> Wonder if we are confident about this change or merge this after branch for some bake time of the change.

I would like to get this into this week's build, I think it works well and we will still have a few days to mess with it.

**Review by chung-lai — APPROVED** [review]
Latest change still look good!

**Comment by evancrow** [comment]
Tests already passed, going to merge to make tonight's TestFlight

**Pull request merged**